src/Features/CSharp/Portable/GoToDefinition/CSharpGoToDefinitionSymbolService.cs

```diff
@@ -119,6 +136,38 @@ protected override ISymbol FindRelatedExplicitlyDeclaredSymbol(ISymbol symbol, C
 
                 return node;
             }
+
+            SyntaxNode? TryFindAccessibleLabel(SyntaxNode? node)
+            {
+                var statement = node?.FirstAncestorOrSelf<GotoStatementSyntax>();
+                if (statement is null)
+                {
+                    return null;
+                }
+
+                var gotoOperation = semanticModel.GetOperation(statement) as IBranchOperation;
+                Debug.Assert(gotoOperation is { BranchKind: BranchKind.GoTo });
+                var target = gotoOperation.Target;
+
+                var expression = statement.Expression;
```

Comment:
i dont' understand why the rest of this is needed.  if you have the target, isn't that an ILabelSymbol?  Why not just return its location?

Reply:
Correct observation, there is no reason for the additional processing; the target is calculated and stored within the IOperation